Shell/SineUtils.cpp

```diff
@@ -252,7 +253,11 @@ SineSymbolExtractor::SymIdIterator SineSymbolExtractor::extractSymIds(Unit* u)
     FormulaUnit* fu=static_cast<FormulaUnit*>(u);
     extractFormulaSymbols(fu->formula(),itms);
   }
-  return pvi( getUniquePersistentIterator(Stack<SymId>::Iterator(itms)) );
+  Stack<SymId> ids(itms.size());
+  DHSet<SymId>::Iterator iter(itms);
+  ids.loadFromIterator(iter);
+  std::sort(ids.begin(), ids.end()); // <- make order deterministic
```

Comment:
I'm not super-familiar with the SInE code - does changing the order we return symbols in break anything? Maybe @quickbeam123 knows.

Reply:
Theoretically, it shouldn't. The definition of relevant symbols in SInE is independent on 'order' ... I'm guessing non-deterministic traversal here would only lead to a non-deterministic performance profile. But @quickbeam123 has looked at SInE more recently than me... :)

Reply:
`SymId` is simply an unsigned. It's nice to care about reproducibility, but unless we hash on pointers, we are deterministic (if a bit chaotic). Unless I am overlooking something, I wouldn't even bother sorting here.